### PR TITLE
added JsonStringLocalizer & added Sass Dart

### PR DESCRIPTION
**Pull request by saeb-panahifar**
List of changes in this pull request.

1. Added JSON Based Localization 
2. Use Dart Sass instead of Sass

### CODE REVIEW

**Comment by vertechcon** [comment]
Both translation files present in Client/Resources (the MainMenu ones) should be in ClientUI.Components/Shared

File: Client/Program.cs [hunk 1]
```diff
@@ -47,9 +52,36 @@
     {
         cp.SetIsOriginAllowed(origin => new Uri(origin).IsLoopback);
     });
-});
-
+});
+
 #region ConfigureServices
+
+builder.Services.AddLocalization(options =>
```
Comment (vertechcon) [inline Client/Program.cs:59]: To support webasm, this configuration should probably be set in ClientUI.Shared/Extensions/ServiceCollectionExtensions

**Review by vertechcon — CHANGES_REQUESTED** [review]
Should move UI associated resources to ClientUI.Components and startup configuration into ClientUI.Shared so we can eventually support webasm mode

**Comment by saeb-panahifar** [comment]
> Both translation files present in Client/Resources (the MainMenu ones) should be in ClientUI.Components/Shared since they're associated with the MainMenu.razor file. This is so we can reuse those resource files when in webasm mode (note just in server mode).


Good point! I moved  `ClientUI/Resources` into `ClientUI.Components/Resources`. Should the resources be in one place (all JSON files in the resources folder) or beside the razor file?

File: Client/Resources/MainMenu-fr.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿{
+  "Dashboard": "Tableau de bord",
+  "About": "Sur"
```
[the remark this answers is not in the document]
Comment (saeb-panahifar) [reply Client/Resources/MainMenu-fr.json:3]: Sorry, what does "À propos" means?

[the remark this answers is not in the document]
Comment (vertechcon) [reply Client/Resources/MainMenu-fr.json:3]: It means "about", there are many different contextual expressions in french to say about something.  We use "À propos du...' or  "à propos de..." to introduce a subject but when nothing follows we just put "À propos". It mostly translates to "On the subject of...".  "Sur" would be used in a context like "I have nothing to say about this" => "Je n'ai rien à dire sur le sujet".

[the remark this answers is not in the document]
Comment (vertechcon) [reply Client/Resources/MainMenu-fr.json:3]: For some reason i had the impression we could highlight code by selecting before adding comments... hmmm

**Comment by vertechcon** [comment]
> In Resources folder (structure 2):
> * 📁 Resources
>   * 📁 MainMenu
>     * `en.json`
>     * `fr.json`

This approach looks pretty good actually! I didn't know we could use subfolders as type names! Makes things a lot cleaner.

**Review by vertechcon — APPROVED** [review]

**Pull request merged**